==== triton/lib/Dialect/TritonGPU/Transforms/Pipeline.cpp#7 - /google/src/cloud/frgossen/fig-export-mlir-rotation-7244-change-4/triton/lib/Dialect/TritonGPU/Transforms/Pipeline.cpp ====
# action=edit type=text
--- triton/lib/Dialect/TritonGPU/Transforms/Pipeline.cpp	2023-04-07 10:46:21.000000000 -0700
+++ triton/lib/Dialect/TritonGPU/Transforms/Pipeline.cpp	2023-04-10 12:34:43.000000000 -0700
@@ -361,7 +361,7 @@
     for (Operation &op : forOp.getLoopBody().front()) {
       if (depOps.contains(&op))
         orderedDeps.push_back(&op);
-      else if (loads.contains(op.getResult(0)))
+      else if (op.getNumResults() > 0 && loads.contains(op.getResult(0)))
         orderedDeps.push_back(&op);
     }
     assert(depOps.size() + loads.size() == orderedDeps.size() &&
@@ -575,7 +575,7 @@
   for (Operation &op : forOp.getLoopBody().front()) {
     if (depOps.contains(&op))
       orderedDeps.push_back(&op);
-    else if (loads.contains(op.getResult(0)))
+    else if (op.getNumResults() > 0 && loads.contains(op.getResult(0)))
       orderedDeps.push_back(&op);
   }
   assert(depOps.size() + loads.size() == orderedDeps.size() &&
